### PR TITLE
Sg 32/packman uninstall error solved

### PR DESCRIPTION
**Pull request by jibola1997**
#ifdef placed in ShotgunMenuitem.cs generation. On uninstall via Packman , the code in that script doesn't compile, eliminating the error, user can then manually delete Shotgun Folder

### CODE REVIEW

File: startup.py [hunk 1]
```diff
@@ -97,7 +97,16 @@ def prepare_launch(self, exec_path, args, file_to_open=None):
...
+        #To be removed when Python Package is integrated properly
+        #Code is passing a PySide2 environment variable to Shotgun unity to use for install
+        #A proper integration would simple Pip install Pyside
+        from sgtk.util.qt_importer import QtImporter
+        pyside_importer = QtImporter()._import_pyside2()
+        pysideFilePath = pyside_importer[2].__path__[0]
+        pysideFilePath=pysideFilePath.replace("Python","Python3")
+        pysideFilePath=os.path.dirname(pysideFilePath)
+        required_env["SHOTGUN_UNITY_PYSIDE_LOCATION"] = pysideFilePath
+
```
Comment (lassond) [inline startup.py:109]: Why is this change showing? I thought you merged that already.
You should rebase/merge your branch to make it disappear

File: plugins/basic/tk_create_menus/generateMenuItems.py [hunk 2]
```diff
@@ -54,7 +56,15 @@ class {className}
...
-    "defineConstraints": []
+    "defineConstraints": [],
+    "versionDefines": [
+        {{
+            "name": "com.unity.integrations.shotgun",
+            "expression": "",
+            "define": "SHOTGUN_INSTALLED"
+        }}
+    ],
+    "noEngineReferences": false
```
Comment (lassond) [inline plugins/basic/tk_create_menus/generateMenuItems.py:67]: what does `noEngineReferences` do?

**Review by lassond — CHANGES_REQUESTED** [review]
I left questions and some change requests

**Comment by lassond** [comment]
I would like to know if you explored what is suggested in the Jira ticket (delete the Shotgun directory on package removed)
Leaving code behind does not look ideal to me. I see @Benoît Hudson suggested that approach in the jira comments, so I guess this is ok

If you leave it like that, I think you should add an explanation. Something like
```
Removing the Shotgun for Unity package leaves this script in the assets/Shotgun directory. It is recommended to manually delete the assets/Shotgun directory after removing the Shotgun for Unity package from a project.
```

File: plugins/basic/tk_create_menus/generateMenuItems.py (hunk 2 above)
Comment (jibola1997) [reply plugins/basic/tk_create_menus/generateMenuItems.py:67]: if it's true , it removes all references to UnityEditor and UnityEngine from that assemble file. I don't think we want that.
Either way, json code added to the asmdef file is added based on the link below. It's automatically generated if you want to define something from the editor.
https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#define-symbols

Comment (lassond) [reply plugins/basic/tk_create_menus/generateMenuItems.py:67]: ok, so you did not add this yourself, it was added automatically (I guess the version of Unity you are using adds that now)

**Pull request merged**